static/app/views/dashboardsV2/gridLayout/dashboard.tsx

```diff
@@ -67,7 +77,36 @@ type Props = {
   onLayoutChange: (layout: Layout[]) => void;
 };
 
-class Dashboard extends Component<Props> {
+type State = {
+  isMobile: boolean;
+  layouts: Layouts;
+};
+
+class Dashboard extends Component<Props, State> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isMobile: false,
+      layouts: {
+        [DESKTOP]: props.layout,
+        [MOBILE]: getMobileLayout(props.layout, props.dashboard.widgets),
+      },
+    };
+  }
+
+  static getDerivedStateFromProps(props, state) {
```

Comment:
I found a bug yesterday when hitting the "Cancel" button :(

Since we only consume the `layout` prop to set the initial state of the grid layout, hitting cancel resets the `layout` in the parent but since the grid layout is already mounted it and it's using its own local state, it doesn't know what to do with this new `layout` it's being passed. What ends up happening is that you cancel and the layout doesn't reset, until you refresh